src/core/Helpers.cpp

```diff
@@ -143,6 +143,25 @@ bool IsGasWorker::operator()(const sc2::Unit& unit_) const {
     return false;
 }
 
+bool IsTownHall::operator()(const sc2::Unit& unit_) const {
```

Comment:
We already have IsTownHall in the API
https://github.com/cpp-sc2/cpp-sc2/blob/2e9f0aeda9da737ed2b1c0551fdad91a8007d5cb/include/sc2api/sc2_unit_filters.h#L30

Please, reuse it.

Reply:
I forgot to rename it -- it should be "IsNonflyingTownhall". I'll fix along with the chaining-filter class.